presto-kafka/src/main/java/io/prestosql/plugin/kafka/encoder/raw/RawRowEncoder.java

```diff
@@ -0,0 +1,375 @@
+                                "No end mapping defined for column '%s' of type '%s'",
+                                columnMapping.getName(),
+                                columnMapping.getType().getDisplayName()));
+                    }
+                }
+            }
+        }
+
+        // make sure the actual/expected lengths match
+        for (ColumnMapping columnMapping : this.columnMappings) {
+            if (columnMapping.getLength() != columnMapping.getFieldType().getSize()) {
+                if (!isVarcharType(columnMapping.getType())) {
+                    throw new IndexOutOfBoundsException(format(
+                            "Mapping length '%s' is not equal to expected length '%s' for column '%s'",
+                            columnMapping.getLength(),
+                            columnMapping.getFieldType().getSize(),
+                            columnMapping.getName()));
+                }
+            }
+        }
+
+        // check that column mappings don't overlap
+        int position = 0;
+        for (ColumnMapping columnMapping : this.columnMappings) {
+            checkArgument(columnMapping.getStart() == position, format(
+                    "Start mapping '%s' for column '%s' does not equal expected mapping '%s'",
+                    columnMapping.getStart(),
+                    columnMapping.getName(),
+                    position));
+            checkArgument(columnMapping.getEnd().isPresent() && columnMapping.getEnd().getAsInt() > columnMapping.getStart(), format(
+                    "End mapping '%s' for column '%s' is less than or equal to start '%s'",
+                    columnMapping.getEnd().getAsInt(),
+                    columnMapping.getName(),
+                    columnMapping.getStart()));
+            position += columnMapping.getLength();
+        }
+
+        this.buffer = ByteBuffer.allocate(position);
+    }
+
+    private class ColumnMapping
+    {
+        private final String name;
+        private final Type type;
+        private final int start;
+        private OptionalInt end;
+        private final FieldType fieldType;
+
+        public ColumnMapping(int index)
+        {
+            EncoderColumnHandle columnHandle = columnHandles.get(index);
+            this.name = columnHandle.getName();
+            this.type = columnHandle.getType();
+            this.fieldType = parseFieldType(columnHandle);
+            checkFieldType(columnHandle, this.fieldType);
+            Optional<String> mapping = Optional.ofNullable(columnHandle.getMapping());
+
+            if (mapping.isPresent()) {
+                Matcher mappingMatcher = MAPPING_PATTERN.matcher(mapping.get());
+                if (!mappingMatcher.matches()) {
+                    throw new IllegalArgumentException(format("Invalid mapping for column '%s'", this.name));
+                }
+                else {
+                    this.start = parseInt(mappingMatcher.group(1));
```

Comment:
wrap with `try`

Maybe extract method `parseOffset()` which does exception handling and call it like:
```
parseOffset(mappingMatcher.group(1), "start", this.name)
parseOffset(mappingMatcher.group(2), "end", this.name)
```